container/lxd/initialisation.go

```diff
@@ -49,6 +76,147 @@ func getPackagingConfigurer(series string) (config.PackagingConfigurer, error) {
 	return config.NewPackagingConfigurer(series)
 }
 
+func configureZFS() {
+	/* create a 100 GB pool by default (sparse, so it won't actually fill
+	 * that immediately)
+	*/
+	output, err := exec.Command(
+		"lxd",
+		"init",
+		"--auto",
+		"--storage-backend", "zfs",
+		"--storage-pool", "lxd",
+		"--storage-create-loop", "100",
+	).CombinedOutput()
+
+	if err != nil {
+		logger.Warningf("configuring zfs failed with %s: %s", err, string(output))
+	}
+}
+
+func configureLXDBridge() error {
+	f, err := os.OpenFile(lxdBridgeFile, os.O_RDWR, 0777)
+	if err != nil {
+		/* We're using an old version of LXD which doesn't have
+		 * lxd-bridge; let's not fail here.
+		 */
+		if os.IsNotExist(err) {
+			logger.Warningf("couldn't find %s, not configuring it", lxdBridgeFile)
+			return nil
+		}
+		return errors.Trace(err)
+	}
+	defer f.Close()
+
+	output, err := exec.Command("ip", "addr", "show").CombinedOutput()
+	if err != nil {
+		return errors.Trace(err)
+	}
+
+	subnet, err := detectSubnet(string(output))
+	if err != nil {
+		return errors.Trace(err)
+	}
+
+	existing, err := ioutil.ReadAll(f)
+	if err != nil {
+		return errors.Trace(err)
+	}
+
+	result := editLXDBridgeFile(string(existing), subnet)
+	_, err = f.Seek(0, 0)
+	if err != nil {
+		return errors.Trace(err)
+	}
+
+	_, err = f.WriteString(result)
+	if err != nil {
+		return errors.Trace(err)
+	}
+
+
+	/* non-systemd systems don't have the lxd-bridge service, so this always fails */
+	_ = exec.Command("service", "lxd-bridge", "restart").Run()
+	return exec.Command("service", "lxd", "restart").Run()
+}
+
+func detectSubnet(ipAddrOutput string) (string, error) {
+	max := 0
+
+	for _, line := range strings.Split(ipAddrOutput, "\n") {
+		trimmed := strings.TrimSpace(line)
+		if trimmed == "" {
+			continue
+		}
+
+		columns := strings.Split(trimmed, " ")
+
+		if len(columns) < 2 {
+			return "", errors.Trace(fmt.Errorf("invalid ip addr output line %s", line))
+		}
+
+		if columns[0] != "inet" {
+			continue
+		}
+
+		addr := columns[1]
+		if !strings.HasPrefix(addr, "10.0.") {
+			continue
+		}
+
+		tuples := strings.Split(addr, ".")
+		if len(tuples) < 4 {
+			return "", errors.Trace(fmt.Errorf("invalid ip addr %s", addr))
+		}
+
+		subnet, err := strconv.Atoi(tuples[2])
+		if err != nil {
+			return "", errors.Trace(err)
+		}
+
+		if subnet > max {
+			max = subnet
+		}
+	}
+
+	return fmt.Sprintf("%d", max+1), nil
+}
+
+func editLXDBridgeFile(input string, subnet string) string {
+	buffer := bytes.Buffer{}
+
+	for _, line := range strings.Split(input, "\n") {
+		out := line
+
+		if strings.HasPrefix(line, "USE_LXD_BRIDGE=") {
```

Comment:
What about if someone has edited the file and these ENV vars don't actually exist (or they've just been commented out)?
Should we have some sort of "found" mapping and insert them at the end? Does the order matter? As it is a "/etc/default" script, I'm guessing they are just env vars?
if so I'd do something like:

// Map the prefixes to the value we want for that line
newValues := map[string]string{
  "USE_LXD_BRIDGE=": `USE_LXD_BRIDGE="true"`,
...
}
found = map[string]bool

for _, line = range ... {
   out := line
   for prefix, value := range newValues {
     if strings.HasPrefix(line, prefix) {
      out = value
      found[prefix] = true
      break
    }
  buffer.Write...
}

for prefix, value := range newValues {
   if !found[prefix] {
     buffer.WriteString(value)
     found[prefix] = true // not necessary but keeps "found" logically consistent
  }
}